apps/examples/wifi_manager/wm_test/wm_test_connectbyrssi_test.c

```diff
@@ -0,0 +1,317 @@
+		strncpy(apconfig.passphrase, ap_info->password, WIFIMGR_PASSPHRASE_LEN);
+		apconfig.passphrase_length = strlen(ap_info->password);
+		if (apconfig.passphrase_length > WIFIMGR_PASSPHRASE_LEN) {
+			apconfig.passphrase[WIFIMGR_PASSPHRASE_LEN + 1] = '\0';
+		} else {
+			apconfig.passphrase[apconfig.passphrase_length + 1] = '\0';
+		}
+		apconfig.ap_crypto_type = ap_info->crypto_type;
+	} else {
+		apconfig.passphrase[0] = '\0';
+		apconfig.passphrase_length = 0;
+		apconfig.ap_crypto_type = ap_info->crypto_type;
+	}
+	wt_print_wifi_ap_profile(&apconfig, "Connecting AP Info");
+	wifi_manager_result_e res = wifi_manager_connect_ap(&apconfig);
+	if (res != WIFI_MANAGER_SUCCESS) {
+		WT_LOGE(TAG, "AP connect failed");
+		return res;
+	}
+	/* Wait for DHCP connection */
+	WT_LOG(TAG, "wait join done");
+	WM_TEST_WAIT;
+	WT_LEAVE;
+	return res;
+}
+
+static void _wt_disconnect()
+{
+	WT_ENTER;
+	wifi_manager_result_e res = WIFI_MANAGER_SUCCESS;
+	/* Disconnect AP */
+	res = wifi_manager_disconnect_ap();
+	if (res != WIFI_MANAGER_SUCCESS) {
+		WT_LOGE(TAG, "Failed to disconnect ap (%d)", res);
+		return;
+	}
+	WM_TEST_WAIT;
+	WT_LEAVE;
+}
+
+
+static void wm_run_connect_by_rssi(void *arg)
+{
+	WT_ENTER;
+	wifi_manager_result_e ret = WIFI_MANAGER_SUCCESS;
+	g_opt = (struct wt_options *)arg;
+	int passed = 0, failed = 0;
+	for (int i = 0; i < g_opt->repeat; i++) {
+		WT_LOG(TAG, "====== Iteration %d starts ======", i + 1);
+		ret = wm_connect_by_rssi_init();
+		if (ret != WIFI_MANAGER_SUCCESS) {
+			WT_LOGE(TAG, "wm_connect_by_rssi_init failed");
+			failed++;
+			goto next_iteration;
+		}
+		memset(g_bssid, '\0', sizeof(g_bssid));
+		memset(g_connected_ap_mac, '\0', sizeof(g_connected_ap_mac));
+		int retry_count = 5;
+		while (retry_count) {
+			ret = _wt_scan();
+			if (ret == WIFI_MANAGER_SUCCESS && g_bssid[0] != '\0')
+				break;
+			// scan till g_bssid get filled or retry count is not got expired
+			sleep(1); // retry after 1 sec sleep
+			retry_count--;
+		}
+		if (0 == retry_count || g_bssid[0] == '\0') {
+			WT_LOGE(TAG, "Scan failed even after multiple retry");
+			failed++;
+			ret = wm_connect_by_rssi_deinit();
+			if (ret != WIFI_MANAGER_SUCCESS) {
+				WT_LOGE(TAG, "wm_connect_by_rssi_deinit failed. so exit");
+				break;
+			}
+			goto next_iteration;
+		}
+
+		// connect to ap
+		retry_count = 5;
+		while (retry_count) {
+			// connect till g_connected_ap_mac get filled or retry count is not got expired
+			ret = _wt_connect(arg);
+			if (ret == WIFI_MANAGER_SUCCESS && g_connected_ap_mac[0] != '\0')
+				break;
+			sleep(1); // retry after 1 sec sleep
+			retry_count--;
+		}
+		if (0 == retry_count) {
+			WT_LOGE(TAG, "Connect ap failed even after multiple retry");
+			failed++;
+			ret = wm_connect_by_rssi_deinit();
+			if (ret != WIFI_MANAGER_SUCCESS) {
+				WT_LOGE(TAG, "wm_connect_by_rssi_deinit failed. so exit");
+				break;
+			}
+			goto next_iteration;
+		} else if (g_connected_ap_mac[0] == '\0') {
```

Comment:
please check in line 258 condition can you append || g_connected_ap_mac[0] == '\0'